Makefile

```diff
@@ -76,31 +96,36 @@ ifeq ($(NVCC),)
     $(info nvcc not found, skipping CUDA builds)
 else
     $(info nvcc found, including CUDA builds)
-    TARGETS += train_gpt2cu test_gpt2cu train_gpt2fp32cu test_gpt2fp32cu
+    TARGETS += train_gpt2cu test_gpt2cu train_gpt2fp32cu test_gpt2fp32cu profile_gpt2cu
 endif
 
 all: $(TARGETS)
 
 train_gpt2: train_gpt2.c
-	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS) $< $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS) $< $(LDLIBS) $(OUTPUT_FILE)
 
 test_gpt2: test_gpt2.c
-	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS) $< $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS) $< $(LDLIBS) $(OUTPUT_FILE)
 
 train_gpt2cu: train_gpt2.cu
 	$(NVCC) $(NVCC_FLAGS) $< $(NVCC_LDFLAGS) -o $@
+	$(WIN_CUDA_RENAME)
 
 train_gpt2fp32cu: train_gpt2_fp32.cu
 	$(NVCC) $(NVCC_FLAGS) $< $(NVCC_LDFLAGS) -o $@
+	$(WIN_CUDA_RENAME)
 
 test_gpt2cu: test_gpt2.cu
 	$(NVCC) $(NVCC_FLAGS) $< $(NVCC_LDFLAGS) -o $@
+	$(WIN_CUDA_RENAME)
 
 test_gpt2fp32cu: test_gpt2_fp32.cu
 	$(NVCC) $(NVCC_FLAGS) $< $(NVCC_LDFLAGS) -o $@
+	$(WIN_CUDA_RENAME)
 
 profile_gpt2cu: profile_gpt2.cu
 	$(NVCC) $(NVCC_FLAGS) -lineinfo $< $(NVCC_LDFLAGS) -o $@
+	$(WIN_CUDA_RENAME)
```

Comment:
Even if the specify  -o to NVCC, it does in classic Window fashion still names its output with the .exe-suffix.
Maybe we should do the same? And stkip the WIN_CUDA_RENAME step. It also looks strange when we are not on Windows to have that line there.

Reply:
Yes, that's what I was trying to fix. The problem is worse if we use the .exe name though. Then all the target names need a suffix. So you end up having something like:

profile_gpt2cu.$(EXE): profile_gpt2.cu 

I thought the above was worse and went with the WIN_CUDA_RENAME macro....